### PR TITLE
fix(cascader): prevent log error for before filter

### PR DESCRIPTION
**Pull request by soc221b**
Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer to relative issues for your PR.

reproduce link: https://codesandbox.io/s/element-plus-spa-forked-ourpv?file=/src/App.vue

reproduce steps: 
1. open console
2. filter and type 'a'

Expected:
There should not log error.

### CODE REVIEW

**Comment by Ryan2128** [comment]
Why need to do this?

**Comment by iendeavor** [comment]
Ummm, Why not?

File: packages/cascader/src/index.vue [hunk 1]
```diff
@@ -504,6 +504,7 @@ export default defineComponent({
 
       if (isPromise(passed)) {
         passed.then(calculateSuggestions)
+          .catch(() => { /* prevent log error */ })
```
Comment (JeremyWuuuuu) [inline packages/cascader/src/index.vue:507]: Why should we hide error from bubbling up to the users?

Comment (JeremyWuuuuu) [reply packages/cascader/src/index.vue:507]: I personally think that the error should always be handled from the upper layer instead of us.

Comment (iendeavor) [reply packages/cascader/src/index.vue:507]: @JeremyWuuuuu 

Users can catch that error in the corresponding callback too:

https://codesandbox.io/s/element-plus-spa-forked-jfhlz?file=/src/App.vue:290-321

In addition, IMHO, user have to write more code (in `onErrorCaptured` hook) to catch that the error bubbled from component, and that would be annoying.

Comment (JeremyWuuuuu) [reply packages/cascader/src/index.vue:507]: Well, point made. If we bubbles the error to the user layer the whole application crashes, which is not something we would like to see.

**Review by jw-foss — APPROVED** [review]

**Pull request merged**